wremnants/syst_tools.py

```diff
@@ -918,6 +918,29 @@ def add_theory_hists(results, df, args, dataset_name, corr_helpers, qcdScaleByHe
         helicity_moments_scale = df.HistoBoost("nominal_gen_helicity_moments_scale", axes, [*cols, "helicity_moments_scale_tensor"], tensor_axes = [axis_helicity, *theory_tools.scale_tensor_axes], storage=hist.storage.Double())
         results.append(helicity_moments_scale)
 
+        # below logic only valid for specific columns
+        if cols == ["massVgen", "absYVgen", "ptVgen", "chargeVgen"]:
+
+            df = df.Define("helicity_moments_scale_lhe_tensor", "wrem::makeHelicityMomentScaleTensor(csSineCosThetaPhilhe, scaleWeights_tensor, nominal_weight)")
```

Comment:
Do we produce all of these all the time? Do we really need to do that?

Reply:
We do, but it shouldn't really hurt anything and the performance impact should be marginal.

Reply:
The lazy evaluation of the graph also means these Defines are never actually called except in the gen histmaker.